lib/double_entry/reporting/aggregate_array.rb

```diff
@@ -50,8 +53,9 @@ def retrieve_aggregates
         scope = LineAggregate.
                 where(:function => function).
                 where(:range_type => 'normal').
-                where(:account => account.to_s).
-                where(:code => code.to_s).
+                where(:account => account.try(:to_s)).
+                where(:code => code.try(:to_s)).
+                where(:partner_account => partner_account.try(:to_s)).
```

Comment:
Duh, please ignore my previous comment. This was a deliberate change to fix an existing bug, check out this commit:
https://github.com/envato/double_entry/commit/536d6faa9bff6b65b8c92e782844f62869afaa34

The issue was that calling `to_s` on `nil` returns an empty string (`""`) which does not find the records saved with a `NULL` value in the database. We believe that this has led to some aggregates with cached values stored in the DB to be re-calculated every time. The spec added in the above commit catches this scenario.